lib/aspect.ts

```diff
@@ -0,0 +1,32 @@
+import { IAspect, IConstruct, App, Stack } from "@aws-cdk/core";
+import { Watchful } from 'cdk-watchful';
+import * as sqs from '@aws-cdk/aws-sqs';
+import * as ddb from '@aws-cdk/aws-dynamodb';
+
+export class WatchAspect implements IAspect {
+  readonly app:App;
+  _watchful: Watchful;
+
+  constructor(scope: App) {
+    this.app = scope;
+  }
+
+  public visit(node: IConstruct): void {
+    if (node instanceof ddb.Table) {
+      const dd = node as ddb.Table;
+      this.watchful.watchDynamoTable('Meerkat table', dd);
+    }
+  }
+
+  get watchful(): Watchful {
+    if (this.watchful === undefined) {
+      const stack = new Stack(this.app, 'watchful-Stack');
```

Comment:
Right. And the big question is now, how is the user going to add this stack to the pipeline?

Reply:
This is to cover the use case of a stack added to the application by the 'code' which means the user will not add it to the pipeline, it will be added in some magical way to the pipeline, the assumption is that once aspects execute we already know the stack dependencies so we should be able to add the aspect created stack to the pipeline on behalf of the user.

Reply:
Hmm. Not sure I buy that. Yes, it would be great if we could do that (later on), but I think we're not aiming for the magic just yet. That's why I would like your example to also include applying this aspect and adding it to the pipeline.

Becomes even more interesting with multiple copies of the same application, because now you have to pay attention where you apply the aspect.